internal/cose.go

```diff
@@ -0,0 +1,165 @@
+package internal
+
+import (
+	"crypto/ecdsa"
+	"crypto/sha512"
+	"fmt"
+	"math/big"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+type SigningAlgorithm int
+
+const (
+	BadOrMissingAlgorithm SigningAlgorithm = iota
+	ES384
+)
+
+// https://datatracker.ietf.org/doc/html/rfc8152#section-4.2
+type CoseSign1 struct {
+	_ struct{} `cbor:",toarray"`
+
+	Protected   []byte
+	Unprotected cbor.RawMessage
+	Payload     []byte
+	Signature   []byte
+}
+
+type CoseHeader struct {
+	Alg interface{} `cbor:"1,keyasint,omitempty" json:"alg,omitempty"`
+}
+
+func MakeCoseSign1FromBytes(bytes []byte) (CoseSign1, error) {
```

Comment:
Where "bytes" are a base64 decoded attestation string. Handles unmarshaling into cose sign 1 struct and checking that required fields exist

Reply:
Be careful with using "bytes" as a variable.  It is a package name in the go standard library see: https://pkg.go.dev/bytes.  (I have been bit a few times by shadowing the package a few times recently.)

Reply:
I suggest we change this to `CoseSign1.UnmarshalBinary` - see my comment in document.go for a discussion